font-codegen/src/parsing.rs

```diff
@@ -1060,20 +1060,13 @@ fn build_type_map(items: &Items) -> HashMap<String, FieldType> {
 fn resolve_ident<'a>(
     known: &'a HashMap<String, FieldType>,
     field_name: &syn::Ident,
-    ident: &syn::Ident,
+    field_type: &syn::Ident,
 ) -> Result<&'a FieldType, syn::Error> {
-    if let Some(item) = known.get(&ident.to_string()) {
-        debug!("Resolve {}: {} to {:?}", field_name, ident, item);
+    if let Some(item) = known.get(&field_type.to_string()) {
+        debug!("Resolve {}: {} to {:?}", field_name, field_type, item);
         Ok(item)
     } else {
-        Err(syn::Error::new(
-            field_name.span(),
-            format!(
-                "I don't know what this is, do you? {}: {}",
-                field_name.to_string().as_str(),
-                ident
-            ),
-        ))
+        Err(syn::Error::new(field_type.span(), "Error: undeclared type"))
```

Comment:
Why are we reducing the error information? - seeing the name *and* type together, formed how you'd likely find them in code, is useful

Reply:
When the error is printed, it will show in the source tree where it occurs, and so the name and type (as well as the file/line number etc) are visible in the final message.

Reply:
to clarify, here is the old message and the new:

old:

```
Error: 
  × parsing failed
    ╭─[resources/codegen_inputs/test.rs:59:1]
 59 │     value: u16,
 60 │     other_thing: ExternalThing,
    ·   ─────┬─────
    ·        ╰── I don't know what this is, do you? other_thing: ExternalThing
 61 │ }
    ╰────
```

new:

```
Error: 
  × parsing failed
    ╭─[resources/codegen_inputs/test.rs:89:1]
 89 │     value: u16,
 90 │     other_thing: ExternalThing,
    ·                ──────┬──────
    ·                      ╰── Error: undeclared type
 91 │ }
    ╰────
```

and as an aside, I do not know why all of the spans are always a few columns out of alignment.. 😒

Reply:
> I do not know why all of the spans are always a few columns out of alignment

At one point I was firmly convinced I had caused that and then couldn't figure out how before realizing it happened even in the original code rofl